### PR TITLE
Use newer charm.v6 repo and update tests

### PR DESCRIPTION
**Pull request by wallyworld**
Update to use newer charm.v6 repo.
Fix a failing test.
Allow tests which use javascript in mongo to be skipped.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/56/
Test FAILed.

File: flag.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+package charmrepo
+
+import (
+	"flag"
+	"os"
+
+	jujutesting "github.com/juju/testing"
+)
+
+var noTestMongoJs *bool = flag.Bool("notest-mongojs", false, "Disable MongoDB tests that require JavaScript")
```
Comment (rogpeppe) [inline flag.go]: What's this doing in a non-test file? It seems entirely test related to me and AFAICS there's no non-test code that uses it.

Comment (rogpeppe) [inline flag.go]: Oh yes, you can omit the '*bool' here too.

File: flag.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+func init() {
+	if os.Getenv("JUJU_NOTEST_MONGOJS") == "1" || jujutesting.MgoServer.WithoutV8 {
```
Comment (rogpeppe) [inline flag.go]: I think let's just use either the env var or the flag. Probably the env var 'cos that means you can go test ./... .

File: flag.go (hunk 1 above)
Comment (wallyworld) [inline flag.go]: I copied this directly from charmstore repo

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/57/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/58/
Test PASSed.

**Pull request closed without merging**